### PR TITLE
Multimaster

### PR DESCRIPTION
**Pull request by tardyp**
some doc about multimaster concepts as per http://trac.buildbot.net/ticket/3395

with some nice block diagrams

### CODE REVIEW

**Comment by djmitche** [comment]
I'm not sure where to see the output.  It looks like this is sending workers through a load balancer to talk to the master -- what would that load balancer be?

**Comment by djmitche** [comment]
Also looks like the spellchecker is complaining..

**Comment by tardyp** [comment]
https://882-760165-gh.circle-artifacts.com/0//home/ubuntu/buildbot/master/docs/_build/html/manual/concepts.html#multimaster

**Comment by tardyp** [comment]
link to new version:
https://892-760165-gh.circle-artifacts.com/0//home/ubuntu/buildbot/master/docs/_build/html/manual/concepts.html#multimaster

**Comment by djmitche** [comment]
Looks good, except I'm still confused about how workers communicate through a load balancer..

**Comment by tardyp** [comment]
@djmitche well, as you told in the original text, any loadbalancing technique can work, which I represent as an intermediate loadbalancer (whether its DNS or haproxy)

**Comment by djmitche** [comment]
OK, I'm a little nervous there that people will say "oh boy looks easy!" and we don't really have any good guides for them in load-balancing workers, but let's solve that problem when it occurs :)

File: master/docs/spelling_wordlist.txt [hunk 1]
```diff
@@ -30,6 +30,11 @@ arguent
 argv
 ascii
 Async
+asymetrics
```
Comment (rutsky) [inline master/docs/spelling_wordlist.txt]: Shouldn't it be "asy**mm**etric"?

File: master/docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -766,6 +773,9 @@ SVN
 svnmailer
 svnurl
 symlinks
+symetric
```
Comment (rutsky) [inline master/docs/spelling_wordlist.txt]: "symmetric"

**Comment by rutsky** [comment]
@tardyp I like how you define diagram  using `blockdiag`, but honestly result looks awful:

blockdiag-b5bf9dfb984e6fc6abca03611e37b1c3fedc9d54

Aliased edges of lines and fonts hurts my eyes :)
Have you tried some options to improve generated results?
For example, I think [images in SVG](http://blockdiag.com/en/blockdiag/sphinxcontrib.html#configuration-file-options) may look significantly better:

```
blockdiag_html_image_format = "SVG"
```

or using other options such as `blockdiag_antialias` may help.

**Comment by tardyp** [comment]
updated diagram with colors from material design palette:
https://900-760165-gh.circle-artifacts.com/0//home/ubuntu/buildbot/master/docs/_build/html/manual/concepts.html#multimaster

image

**Comment by rutsky** [comment]
@tardyp looks great, thanks!

**Pull request merged**